update vignette for extending boards for using in new R package (#291)
rstudio/pins: vignettes/boards-extending.Rmd

  unlink(file.path("pins", name), recursive = TRUE)
}
```

> If we want to provide the above methods in a separate package, this new package also needs to export the s3 generics from `pins`. For example:

Reviewer: ```suggestion
To extend `pins` in a separate package, make sure to export the s3 generics from `pins`. For example:
```
